### PR TITLE
improvement: reduce io operation when load appfile

### PR DESCRIPTION
**Pull request by dylandee**

### CODE REVIEW

File: pkg/serverlib/application.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package serverlib
 import (
 	"bytes"
 	"context"
+	j "encoding/json"
```
Comment (wonderflow) [inline pkg/serverlib/application.go:6]: I think `json` is better than `j` here

File: pkg/serverlib/application.go [hunk 2]
```diff
@@ -327,20 +329,34 @@ func GetServicesWhenDescribingApplication(cmd *cobra.Command, app *api.Applicati
...
+	switch ext {
+	case ".json":
 		dest = "vela.json"
-	} else if ext == ".yaml" || ext == ".yml" {
+		af, err = api.JSONToYaml(body, af)
+	case ".yaml", ".yml":
+		dest = "vela.yaml"
+		err = yaml.Unmarshal(body, af)
+	default:
 		dest = "vela.yaml"
```
Comment (wonderflow) [inline pkg/serverlib/application.go:348]: Does it should be `Appfile` ?

File: pkg/serverlib/application.go [hunk 3]
```diff
@@ -327,20 +329,33 @@ func GetServicesWhenDescribingApplication(cmd *cobra.Command, app *api.Applicati
 	return targetServices, nil
 }
 
-func saveRemoteAppfile(url string) (string, error) {
+func saveAndLoadRemoteAppfile(url string) (string, *api.AppFile, error) {
```
Comment (wonderflow) [inline pkg/serverlib/application.go:332]: if we already return the AppFile, we don't need the filePath as return value

**Review by wonderflow — APPROVED** [review]
LGTM after the import header changed

**Review by dylandee — COMMENTED** [review]
> LGTM after the import header changed

'json' is used below for package '"k8s.io/apimachinery/pkg/runtime/serializer/json"'

what about 'json2' or 'buildinjson' ?

**Comment by wonderflow** [comment]
> > LGTM after the import header changed
> 
> 'json' is used below for package '"k8s.io/apimachinery/pkg/runtime/serializer/json"'
> 
> what about 'json2' or 'buildinjson' ?

I didn't see it was occupied in this file.

**Comment by dylandee** [comment]
> > > LGTM after the import header changed
> > 
> > 
> > 'json' is used below for package '"k8s.io/apimachinery/pkg/runtime/serializer/json"'
> > what about 'json2' or 'buildinjson' ?
> 
> I didn't see it was occupied in this file.

Im confused..Are we discussing the same thing? 😵
image

**Comment by wonderflow** [comment]
@dylandee Oh, sorry, I made the mistake

**Pull request merged**